src/DDTrace/Integrations/Drupal/DrupalIntegration.php

```diff
@@ -51,6 +53,78 @@ public function init()
 
         $integration = $this;
 
+        \DDTrace\hook_function('drupal_bootstrap', function () use ($integration, $rootSpan) {
+            if (
+                $integration->drupalVersion ||
+                !defined('DRUPAL_CORE_COMPATIBILITY') ||
+                DRUPAL_CORE_COMPATIBILITY !== '7.x'
+            ) {
+                return false;
+            }
+
+            // Only load integration once
+            $integration->drupalVersion = DRUPAL_CORE_COMPATIBILITY;
+            $this->drupal7($rootSpan);
+        });
+
+        $this->drupalSymfony($rootSpan);
+
+        return Integration::LOADED;
+    }
+
+    protected function drupal7($rootSpan)
+    {
+        $integration = $this;
+
+        // Trace some methods
+        $methods = array(
+            '_drupal_bootstrap_full', '_drupal_bootstrap_page_cache', '_drupal_bootstrap_database',
+            '_drupal_bootstrap_variables', 'drupal_session_initialize', '_drupal_bootstrap_page_header',
+            'drupal_language_initialize', 'menu_execute_active_handler', 'drupal_deliver_page'
+        );
+        foreach ($methods as $method) {
+            \DDTrace\trace_function($method, function (SpanData $span) use ($method) {
+                $span->name = $method;
+                $span->meta[Tag::COMPONENT] = 'drupal';
+            });
+        }
+
+        // Extract route
+        \DDTrace\hook_function('menu_get_item', null, function ($args, $retval) use ($integration, $rootSpan) {
+            if ($args[0] !== null) {
+                return;
+            }
+
+            if ($integration->shouldRenameRootSpan()) {
+                $path = $retval['path'];
+                $rootSpan->overwriteOperationName($path);
+            }
+        });
+
+        // Can't directly trace functions called by set_error_handler & set_exception_handler
+        \DDTrace\trace_function('_drupal_error_handler_real', function (SpanData $span, $args) {
+            $span->name = '_drupal_error_handler';
+            $span->meta[Tag::COMPONENT] = 'drupal';
+            $span->meta[Tag::ERROR_MSG] = $args[1];
+            $span->meta[TAG::ERROR_TYPE] = 'error handler';
```

Comment:
Any reason why this is `TAG` versus `Tag`?

Reply:
🤷  fixed